infrastructure/async/src/main/java/tech/pegasys/teku/infrastructure/async/eventthread/AsyncRunnerEventThread.java

```diff
@@ -92,7 +95,7 @@ public <T> SafeFuture<T> executeFuture(Supplier<SafeFuture<T>> task) {
   @Override
   public void execute(final Runnable task) {
     if (!started.get()) {
-      return;
+      throw new IllegalStateException("EventThread not started");
```

Comment:
This will cause a lot of noise at shutdown because `started` is set back to false on shutdown but we may still be sending tasks to the executor which we deliberately want to ignore.

Reply:
Oops. Good call. I didn't think of that. I'll revert.

Reply:
Added a comment there that's essentially a copy of your message; hope that's fine. Most of the other methods in that class will throw an exception if the thread isn't started; those are okay right?

Reply:
Yeah those other ones return a failed future which is much more likely to be handled fairly smoothly. I do wonder if we should throw `RejectedExecutionException` rather than `IllegalStateException` but will leave it for now since it doesn't seem to be an issue.